itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestActivePassiveHA.java

```diff
@@ -665,7 +666,18 @@ private String sendAuthMethod(HttpRequestBase method, boolean enableAuth, boolea
 
       httpResponse = client.execute(method);
 
-      return EntityUtils.toString(httpResponse.getEntity());
+      if (httpResponse != null) {
+        StatusLine statusLine = httpResponse.getStatusLine();
+        if (statusLine != null) {
+           response = httpResponse.getStatusLine().getReasonPhrase();
+           statusCode = httpResponse.getStatusLine().getStatusCode();
+          if (statusCode == 200) {
+            return EntityUtils.toString(httpResponse.getEntity());
+          }
```

Comment:
Why this is needed only for HTTP 200? For any other response code this function will return `response`, which is actually `httpResponse.getStatusLine().getReasonPhrase()`. I don't get why this is needed.

Reply:
Only HTTP 200 contains response body other than this there is no/empty response body. If we do not handle this 
testManualFailoverUnauthorized() method  fails at   assertTrue(sendDelete(url1, false).contains("Unauthorized"));
as it tries to get "Unauthorized" string from response body